File: testing/java/ObStack.java
//package iProlog;
import java.util.ArrayList;

class ObStack<T> extends ArrayList<T> {

  private static final long serialVersionUID = 1L;

  final T pop() {
    final int last = this.size() - 1;
    return this.remove(last);
  }

  final void push(final T O) {
    add(O);
  }

  final T peek() {
    return get(this.size() - 1);
  }
public static void main(final String[] args) {
    ObStack s=new ObStack();
s.push(5);
System.out.println(s.peek());
s.push(6);


System.out.println(s.peek());

  }

}

File: testing/cpp/ObStack.cpp

#include<iostream>
#include <vector>

template<typename T>
	class ObStack : public std::vector<T>
	{

  private:
	//  static long serialVersionUID = 1LL;

  public:
	  T pop() //removing elements from the stack
	  {
		int last = this->size() - 1;
		return this->remove(last);
	  }

	  void push(T const O) //inserting elements from the stack
	  {
		this->push_back(O);
	  }

	  T peek()
	  {
		return this->at(this->size() - 1);
	  }
	};

int main()
{
ObStack <int>o; //defining a variable of obstack of type integer
o.push(5); //inserting values
std::cout<<o.peek()<<std::endl; //displaying
o.push(6); //inserting values
std::cout<<o.peek()<<std::endl; //displaying
}


